src/main/java/jarvis/ui/ExpandedLessonCard.java

```diff
@@ -0,0 +1,136 @@
+    private Label id;
+    @FXML
+    private ImageView checkbox;
+    @FXML
+    private Label lessonType;
+    @FXML
+    private Label lessonDesc;
+    @FXML
+    private Label timePeriod;
+    @FXML
+    private Label generalNotes;
+    @FXML
+    private TableView<Student> tableView;
+    @FXML
+    private TableColumn<Student, Integer> studentIndex;
+    @FXML
+    private TableColumn<Student, String> studentNames;
+    @FXML
+    private TableColumn<Student, String> studentAttendance;
+    @FXML
+    private TableColumn<Student, Integer> studentParticipation;
+    @FXML
+    private TableColumn<Student, String> studentNotes;
+
+    private Image tick = new Image(getClass().getResourceAsStream("/images/tick.png"));
+    private Image cross = new Image(getClass().getResourceAsStream("/images/cross.png"));
+
+    /**
+     * Creates an {@code ExpandedLessonCard} with the given {@code Lesson} and index to display.
+     */
+    public ExpandedLessonCard(Lesson lesson, int displayedIndex) {
+        super(FXML);
+        this.lesson = lesson;
+        id.setText(displayedIndex + ". ");
+        if (lesson.isCompleted()) {
+            checkbox.setImage(tick);
+        } else {
+            checkbox.setImage(cross);
+        }
+        switch(lesson.getLessonType()) {
+        case MASTERY_CHECK:
+            lessonType.setText("Mastery Check");
+            break;
+        case CONSULT:
+            lessonType.setText("Consult");
+            break;
+        case STUDIO:
+            lessonType.setText("Studio");
+            // Set participation column for studio
+            studentParticipation.setVisible(true);
+            studentParticipation.setCellValueFactory(s -> new ReadOnlyObjectWrapper<>((
+                    (Studio) lesson).getParticipationForStudent(s.getValue())));
+            break;
+        default:
+            assert false : "There are only 3 types of lesson";
+        }
+        if (lesson.hasDesc()) {
+            lessonDesc.setText("Description: " + lesson.getDesc().lessonDesc);
+        } else {
+            lessonDesc.setText("{No description}");
+        }
+        timePeriod.setText(lesson.getTimePeriod().toString());
+        generalNotes.setText(lesson.getGeneralNotes());
+
+        ObservableList<Student> list = FXCollections.observableArrayList(lesson.getAttendance().getAllStudents());
+        tableView.setPrefHeight(LESSON_CARD_BASE_HEIGHT + list.size() * TABLE_VIEW_ROW_HEIGHT);
+        studentIndex.setCellValueFactory(s -> new ReadOnlyObjectWrapper<>(
+                tableView.getItems().indexOf(s.getValue()) + 1));
+        studentNames.setCellValueFactory(s -> new ReadOnlyStringWrapper(s.getValue().getName().toString()));
+        studentAttendance.setCellValueFactory(s -> new ReadOnlyStringWrapper(lesson.isPresent(s.getValue())));
+        studentNotes.setCellValueFactory(s -> new ReadOnlyStringWrapper(
+                lesson.getStudentNotes(s.getValue())));
+        tableView.setItems(list);
+    }
```

Comment:
I am not too sure but I think this method might be a bit too long, was wondering if splitting it to different methods would be better?